demo/midgard-sdk/src/tx-builder/user-events/deposit/index.ts

```diff
@@ -1 +1,65 @@
+import { Data, Integer, LucidEvolution, Script, TxBuilder} from "@lucid-evolution/lucid";
+import { DepositDatum, DepositInfo, MintRedeemer } from "./types.js";
+import { POSIXTime } from "@/tx-builder/common.js";
+import { sha3_256 } from "@noble/hashes/sha3";
+import { Int } from "effect/Schema";
+
+
+export type DepositParams = {
+  depositScriptAddress: string,
+  mintingPolicy: Script,
+  policyId: string,
+  depositInfo: DepositInfo,
+  inclusionTime: POSIXTime,
+};
+
+/**
+ * Deposit
+ *
+ * @param lucid - The LucidEvolution
+ * @param params - The parameters
+ * @returns {TxBuilder} A TxBuilder instance that can be used to build the transaction.
+ */
+
+export const depositTxBuilder = async (
+  lucid: LucidEvolution,
+  params: DepositParams,
+): Promise<TxBuilder> => {
```

Comment:
Do you think you can use `effect`? Other endpoints can help, like [the commitment one](https://github.com/Anastasia-Labs/midgard/blob/main/demo/midgard-sdk/src/tx-builder/state-queue/commit-block-header.ts).
```suggestion
): Effect.Effect<TxBuilder, HashingError | LucidError> => {
```